rakelib/contact_info_via_veteran_icn.rake

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+namespace :veteran_contact_info do
+  desc 'Build a CSV with contact information for a batch of veteran icns'
+  task :build_contact_info_csv, %i[csv_filename icn_list] => :environment do |_, args|
+    def build_address(mpi_address)
+      line1 = mpi_address.street
+      line2 = mpi_address.street2
+      street = line1
+      street += ", #{line2}" if line2.present?
+      city = mpi_address.city
+      postal_code = mpi_address.postal_code
+      country = mpi_address.country
+
+      "#{street}, #{city} #{postal_code}, #{country}"
+    end
+
+    def build_common_name(mpi_profile)
+      first_name = mpi_profile.given_names[0]
+      middle_name = mpi_profile.given_names[1]
+      last_name = mpi_profile.family_name
+      suffix = mpi_profile.suffix
+
+      [first_name, middle_name, last_name, suffix].compact.join(' ')
+    end
+
+    def retrieve_current_email(vet360_profile)
+      current_emails = vet360_profile.emails.select { |email| email.effective_end_date.nil? }
+
+      raise 'All emails expired!' if current_emails.length.zero?
+
+      current_emails.first.email_address
+    end
+
+    def upload_to_s3(_csv, csv_filename, csv_path, url_life_length: 1.week.to_i)
+      s3_settings = Settings.decision_review.s3
+
+      begin
+        # Do we need a separate bucket for this file upload?
```

Comment:
I don't know how encapsulated the buckets should be as far as what they contain, but I'm just using the one set up for Decision Reviews (which seems to be generally for evidence uploads, I think?)

Reply:
We've reused buckets on our other reporting tasks. I feel it is ok to do so here as well.